tslearn/metrics.py

```diff
@@ -12,14 +12,77 @@
 from sklearn.utils import check_random_state
 from tslearn.soft_dtw_fast import _soft_dtw, _soft_dtw_grad, \
     _jacobian_product_sq_euc
+from tslearn.cysax import cydist_sax
 
 from tslearn.utils import to_time_series, to_time_series_dataset, ts_size, \
     check_equal_size
 
 __author__ = 'Romain Tavenard romain.tavenard[at]univ-rennes2.fr'
 
 GLOBAL_CONSTRAINT_CODE = {None: 0, "": 0, "itakura": 1, "sakoe_chiba": 2}
-VARIABLE_LENGTH_METRICS = ["dtw", "gak", "softdtw"]
+VARIABLE_LENGTH_METRICS = ["dtw", "gak", "softdtw", "sax"]
```

Comment:
Is SAX really supposed to be a variable-length metric? Isn't is supposed to run on equal size time series?

Reply:
Hmm that is a good call of which I am not entirely sure... But since SAX converts a longer timeseries to a fixed number of segments, it could maybe work on variable length timeseries as well? I would have to do a bit more research on this to see whether that makes sense...

Reply:
So I think SAX can handle variable lengths:
* The timeseries are converted to a fixed number of segments
* In the original paper (https://www.cs.ucr.edu/~eamonn/SAX.pdf) they state: "SAX allows a time series of arbitrary length n to be reduced to a string  of  arbitrary  length w"
* Dr. Lin uses SAX as pre-processing step in another work (https://arxiv.org/pdf/1802.04883.pdf) titled "Efficient Discovery of Variable-length Time SeriesMotifs with Large Length Range in Million ScaleTime Series"


Perhaps we could put a test in place for these variable-length metrics?

Reply:
Had to make some changes in PAA to make this work. It is no longer vectorized anymore though:
https://github.com/rtavenar/tslearn/pull/152/commits/b96ebe95e2901cdb27b8c7ea5bead2c0b9265e59#diff-a071054ac597f00ec51b4bf94e99d23fR143

Perhaps this can be written more efficiently to avoid all these for-loops?